pkg/client/sql.go

```diff
@@ -276,7 +276,7 @@ func verifyRowAgainst(row *schema.Row, decodedRow map[uint32]*schema.SQLValue, c
 }
 
 func decodeRow(encodedRow []byte, colTypes map[uint32]sql.SQLValueType) (map[uint32]*schema.SQLValue, error) {
-	off := 0
+	off := 1 // ignore DELETED flag during decoding
```

Comment:
Hmm, this introduces compatibility breakage in the client library. It will come with surprising results if the client is using 1.1 and the server is 1.2. If that's the case then we should definitely deprecate the old version so that people are warned about this.